libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1075,19 +1075,24 @@ void AP_InertialSensor::wait_for_sample(void)
     if (!_hil_mode) {
         // we also wait for at least one backend to have a sample of both
         // accel and gyro. This normally completes immediately.
-        bool gyro_available = false;
-        bool accel_available = false;
-        while (!gyro_available || !accel_available) {
+        while (true) {
+            bool gyro_available = false;
+            bool accel_available = false;
```

Comment:
> The intention is to exit the loop when we have both set, but they won't necessarily be set on the same run of the loop

They actually will, because of the following:
```c++
              for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
                  gyro_available |= _new_gyro_data[i];
                  accel_available |= _new_accel_data[i];
              }
```

But, looking again at the code, I'm initializing them on every iteration. Now I have mixed feelings and am towards leaving the declaration and initialization outside the loop.